.travis.yml

```diff
@@ -43,6 +75,11 @@ before_script:
   - python ci/install_coverage_subprocess_pth.py
 script:
   - COVERAGE_PROCESS_START="$TRAVIS_BUILD_DIR/.coveragerc" PYTHONPATH='.:tests' pytest -r s
+  - if [[ $PROJECT != "" ]]; then
+        pushd ../$PROJECT;
+        python -mpytest -vl;
```

Comment:
the fact that you have `;` to chain the commands is probably hidding the return pytest code.You should probably do try something like:

```bash
if [[ $PROJECT != "" ]]; then
    pushd ../$PROJECT;
    python -mpytest -vl;
    TEST_RETURN_CODE = $?;
    popd;
    if [[ "$TEST_RETURN_CODE" != "0" ]]; then exit $TEST_RETURN_CODE; fi
fi
```